core/src/main/java/org/apache/carbondata/core/carbon/datastore/SegmentTaskIndexStore.java

```diff
@@ -82,23 +85,24 @@ public SegmentTaskIndexStore(String carbonStorePath, CarbonLRUCache lruCache) {
   @Override
   public SegmentTaskIndexWrapper get(TableSegmentUniqueIdentifier tableSegmentUniqueIdentifier)
       throws IOException {
-    SegmentTaskIndexWrapper segmentTaskIndexWrapper =
-        loadAndGetTaskIdToSegmentsMap(tableSegmentUniqueIdentifier.getSegmentToTableBlocksInfos(),
-            tableSegmentUniqueIdentifier.getAbsoluteTableIdentifier(),
-            tableSegmentUniqueIdentifier);
+    SegmentTaskIndexWrapper segmentTaskIndexWrapper = null;
+    try {
+      segmentTaskIndexWrapper =
+          loadAndGetTaskIdToSegmentsMap(tableSegmentUniqueIdentifier.getSegmentToTableBlocksInfos(),
+              tableSegmentUniqueIdentifier.getAbsoluteTableIdentifier(),
+              tableSegmentUniqueIdentifier);
+    } catch (IndexBuilderException e) {
+      throw new IOException(e.getMessage(), e);
+    } catch (Throwable e) {
+      throw new IOException("Problem in loading segment block.", e);
+    }
+
     if (null != segmentTaskIndexWrapper) {
       segmentTaskIndexWrapper.incrementAccessCount();
```

Comment:
Please remove the segmentTaskIndexWrapper.incrementAccessCount();
It is not needed here.